components/ai_chat/ai_chat_tab_helper.cc

```diff
@@ -279,8 +298,8 @@ void AIChatTabHelper::DistillViaAlgorithm(const ui::AXTree& tree) {
 void AIChatTabHelper::CleanUp() {
   chat_history_.clear();
   article_summary_.clear();
-  SetRequestInProgress(false);
```

Comment:
Removing this line from cleanup deepens the issue where navigating away does not clean up some state. Right now, that problem manifests by any in-progress requests showing their result on another page. Now that problem will also have the UI show as in-progress. If we're going to have `AIChatAPI` store state and be single-request-only, then perhaps it's time in this PR to add a `Cancel()` method to it.